MessageDatabase_Impl does not exist

` E  FATAL EXCEPTION: main
                                                                                                    Process: com.devedroy.contactsapp, PID: 15295
                                                                                                    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.devedroy.contactsapp/com.devedroy.contactsapp.SendMessageActivity}: java.lang.RuntimeException: Cannot find implementation for com.devedroy.contactsapp.data.MessageDatabase. MessageDatabase_Impl does not exist
                                                                                                    	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3771)
                                                                                                    	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3938)
                                                                                                    	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2291)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:210)
                                                                                                    	at android.os.Looper.loop(Looper.java:299)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8319)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:556)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1038)
                                                                                                    Caused by: java.lang.RuntimeException: Cannot find implementation for com.devedroy.contactsapp.data.MessageDatabase. MessageDatabase_Impl does not exist
                                                                                                    	at androidx.room.Room.getGeneratedImplementation(Room.kt:58)
                                                                                                    	at androidx.room.RoomDatabase$Builder.build(RoomDatabase.kt:1356)
                                                                                                    	at com.devedroy.contactsapp.SendMessageActivity.onCreate(SendMessageActivity.kt:48)
                                                                                                    	at android.app.Activity.performCreate(Activity.java:8163)
                                                                                                    	at android.app.Activity.performCreate(Activity.java:8130)
                                                                                                    	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1330)
                                                                                                    	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3744)
                                                                                                    	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3938) 
                                                                                                    	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103) 
                                                                                                    	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
                                                                                                    	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2291) 
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106) 
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:210) 
                                                                                                    	at android.os.Looper.loop(Looper.java:299) 
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8319) 
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method) 
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:556) 
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1038) `

--- a/.idea/compiler.xml
+++ b/.idea/compiler.xml
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
   <component name="CompilerConfiguration">
-    <bytecodeTargetLevel target="18" />
+    <bytecodeTargetLevel target="11" />
   </component>
 </project>

--- a/.idea/gradle.xml
+++ b/.idea/gradle.xml
@@ -7,6 +7,7 @@
         <option name="testRunner" value="GRADLE" />
         <option name="distributionType" value="DEFAULT_WRAPPED" />
         <option name="externalProjectPath" value="$PROJECT_DIR$" />
+        <option name="gradleJvm" value="Embedded JDK" />
         <option name="modules">
           <set>
             <option value="$PROJECT_DIR$" />

--- a/.idea/misc.xml
+++ b/.idea/misc.xml
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
   <component name="ExternalStorageConfigurationManager" enabled="true" />
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_18" default="true" project-jdk-name="18" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_11" default="true" project-jdk-name="11" project-jdk-type="JavaSDK">
     <output url="file://$PROJECT_DIR$/build/classes" />
   </component>
   <component name="ProjectType">

--- a/.idea/vcs.xml
+++ b/.idea/vcs.xml
@@ -1,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$" vcs="Git" />
+  </component>
+</project>

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -1,6 +1,7 @@
 plugins {
     id 'com.android.application'
     id 'org.jetbrains.kotlin.android'
+    id 'kotlin-kapt'
 }
 
 android {
@@ -23,11 +24,11 @@ android {
         }
     }
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
     }
     kotlinOptions {
-        jvmTarget = '1.8'
+        jvmTarget = '11'
     }
     buildFeatures {
         viewBinding true
@@ -56,6 +57,6 @@ dependencies {
     def room_version = "2.5.1"
     implementation "androidx.room:room-runtime:$room_version"
     implementation "androidx.room:room-ktx:$room_version"
-    annotationProcessor "androidx.room:room-compiler:$room_version"
+    kapt "androidx.room:room-compiler:$room_version"
 
 }

--- a/app/src/main/java/com/devedroy/contactsapp/message/MessagesFragment.kt
+++ b/app/src/main/java/com/devedroy/contactsapp/message/MessagesFragment.kt
@@ -44,7 +44,7 @@ class MessagesFragment : Fragment() {
         }
     }
 
-    private suspend fun getMessages(): List<MessageModel> = withContext(Dispatchers.Main) {
+    private suspend fun getMessages(): List<MessageModel> = withContext(Dispatchers.IO) {
         App.messageDatabase.messageDao().getMessages()
     }
 

--- a/build.gradle
+++ b/build.gradle
@@ -1,6 +1,6 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 plugins {
-    id 'com.android.application' version '8.0.0' apply false
-    id 'com.android.library' version '8.0.0' apply false
+    id 'com.android.application' version '7.4.2' apply false
+    id 'com.android.library' version '7.4.2' apply false
     id 'org.jetbrains.kotlin.android' version '1.8.0' apply false
 }

